### ISSUE TITLE
Can't dup NilClass

### ISSUE THREAD

**Issue by toobulkeh** [vcr/vcr#521]
My first time using VCR, so please excuse my ignorance.

We're getting this error with a pretty minimal setup when trying to make a call to Google Maps for the geocoder gem.

Setup:
```
VCR.configure do |config|
  config.cassette_library_dir = Rails.root.join("spec", "vcr")
  config.allow_http_connections_when_no_cassette = true
  config.ignore_localhost = true
  config.hook_into :webmock

  config.around_http_request do |request|
    case request.uri
    when /maps.googleapis.com/, /api.twilio.com/
      uri = URI(request.uri)
      name = "#{[uri.host, uri.path, uri.query, request.method].join('/')}"
      VCR.use_cassette(name, &request)
    else
      VCR.turned_off(&request)
    end
  end
end
```

Error and Backtrace:
```
Failures:

  1) Contractor Sign Up New contractor creates organization
     Got 1 failure and 1 other error:

     1.1) Failure/Error: expect(current_path).to eq(contractor_dashboard_path)
            
            expected: "/c/dashboard"
                 got: "/c/organizations"
            
            (compared using ==)
          # ./spec/features/contractors/signup_spec.rb:38:in `block (2 levels) in <top (required)>'

     1.2) Failure/Error: Unable to find matching line from backtrace
          TypeError:
            can't dup NilClass
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/vcr-3.0.0/lib/vcr.rb:379:in `dup'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/vcr-3.0.0/lib/vcr.rb:379:in `dup_context'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/vcr-3.0.0/lib/vcr.rb:364:in `get_context'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/vcr-3.0.0/lib/vcr.rb:355:in `current_context'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/vcr-3.0.0/lib/vcr.rb:369:in `context_value'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/vcr-3.0.0/lib/vcr.rb:389:in `cassettes'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/vcr-3.0.0/lib/vcr.rb:49:in `current_cassette'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/vcr-3.0.0/lib/vcr/request_handler.rb:101:in `request_summary'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/vcr-3.0.0/lib/vcr/request_handler.rb:7:in `handle'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/vcr-3.0.0/lib/vcr/library_hooks/webmock.rb:127:in `block in <module:WebMock>'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/stub_registry.rb:28:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/stub_registry.rb:28:in `block in register_global_stub'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/request_pattern.rb:36:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/request_pattern.rb:36:in `matches?'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/stub_registry.rb:58:in `block in request_stub_for'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/stub_registry.rb:57:in `each'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/stub_registry.rb:57:in `detect'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/stub_registry.rb:57:in `request_stub_for'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/stub_registry.rb:50:in `response_for_request'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/http_lib_adapters/net_http.rb:79:in `request'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/geocoder-1.2.11/lib/geocoder/lookups/base.rb:284:in `block in make_api_request'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/http_lib_adapters/net_http.rb:123:in `start_without_connect'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/webmock-1.21.0/lib/webmock/http_lib_adapters/net_http.rb:150:in `start'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/geocoder-1.2.11/lib/geocoder/lookups/base.rb:276:in `make_api_request'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/geocoder-1.2.11/lib/geocoder/lookups/base.rb:226:in `fetch_raw_data'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/geocoder-1.2.11/lib/geocoder/lookups/base.rb:177:in `fetch_data'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/geocoder-1.2.11/lib/geocoder/lookups/google.rb:37:in `results'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/geocoder-1.2.11/lib/geocoder/lookups/base.rb:47:in `search'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/geocoder-1.2.11/lib/geocoder/query.rb:11:in `execute'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/geocoder-1.2.11/lib/geocoder.rb:22:in `search'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/geocoder-1.2.11/lib/geocoder/stores/base.rb:100:in `do_lookup'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/geocoder-1.2.11/lib/geocoder/stores/active_record.rb:270:in `geocode'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:432:in `block in make_lambda'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:239:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:239:in `block in halting'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:506:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:506:in `block in call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:506:in `each'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:506:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:92:in `__run_callbacks__'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:778:in `_run_validation_callbacks'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activemodel-4.2.4/lib/active_model/validations/callbacks.rb:113:in `run_validations!'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activemodel-4.2.4/lib/active_model/validations.rb:337:in `valid?'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/validations.rb:58:in `valid?'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/validations.rb:83:in `perform_validations'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/validations.rb:37:in `save'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/attribute_methods/dirty.rb:21:in `save'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/transactions.rb:286:in `block (2 levels) in save'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/transactions.rb:351:in `block in with_transaction_returning_status'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `block in transaction'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/connection_adapters/abstract/transaction.rb:184:in `within_new_transaction'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/transactions.rb:220:in `transaction'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/transactions.rb:348:in `with_transaction_returning_status'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/transactions.rb:286:in `block in save'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/transactions.rb:301:in `rollback_active_record_state!'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/transactions.rb:285:in `save'
          # ./app/controllers/contractors/organizations_controller.rb:15:in `create'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_controller/metal/implicit_render.rb:4:in `send_action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/abstract_controller/base.rb:198:in `process_action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_controller/metal/rendering.rb:10:in `process_action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/abstract_controller/callbacks.rb:20:in `block in process_action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:117:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:117:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:555:in `block (2 levels) in compile'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:505:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:505:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:92:in `__run_callbacks__'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:778:in `_run_process_action_callbacks'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:81:in `run_callbacks'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/abstract_controller/callbacks.rb:19:in `process_action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_controller/metal/rescue.rb:29:in `process_action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/notifications.rb:164:in `block in instrument'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/notifications.rb:164:in `instrument'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_controller/metal/instrumentation.rb:30:in `process_action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/railties/controller_runtime.rb:18:in `process_action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/abstract_controller/base.rb:137:in `process'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionview-4.2.4/lib/action_view/rendering.rb:30:in `process'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_controller/metal.rb:196:in `dispatch'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_controller/metal/rack_delegation.rb:13:in `dispatch'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_controller/metal.rb:237:in `block in action'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/routing/route_set.rb:76:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/routing/route_set.rb:76:in `dispatch'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/routing/route_set.rb:45:in `serve'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/journey/router.rb:43:in `block in serve'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/journey/router.rb:30:in `each'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/journey/router.rb:30:in `serve'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/routing/route_set.rb:821:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/warden-1.2.3/lib/warden/manager.rb:35:in `block in call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/warden-1.2.3/lib/warden/manager.rb:34:in `catch'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/warden-1.2.3/lib/warden/manager.rb:34:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/etag.rb:24:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/conditionalget.rb:38:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/head.rb:13:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/middleware/params_parser.rb:27:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/middleware/flash.rb:260:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/session/abstract/id.rb:225:in `context'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/session/abstract/id.rb:220:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/middleware/cookies.rb:560:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/query_cache.rb:36:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activerecord-4.2.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:88:in `__run_callbacks__'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:778:in `_run_call_callbacks'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/callbacks.rb:81:in `run_callbacks'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/middleware/callbacks.rb:27:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/airbrake-4.3.1/lib/airbrake/rails/middleware.rb:13:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/railties-4.2.4/lib/rails/rack/logger.rb:38:in `call_app'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/railties-4.2.4/lib/rails/rack/logger.rb:20:in `block in call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/tagged_logging.rb:68:in `block in tagged'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/tagged_logging.rb:26:in `tagged'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/tagged_logging.rb:68:in `tagged'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/railties-4.2.4/lib/rails/rack/logger.rb:20:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/middleware/request_id.rb:21:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/methodoverride.rb:22:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/runtime.rb:18:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/activesupport-4.2.4/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/lock.rb:17:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/actionpack-4.2.4/lib/action_dispatch/middleware/static.rb:116:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/sendfile.rb:113:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/airbrake-4.3.1/lib/airbrake/user_informer.rb:16:in `_call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/airbrake-4.3.1/lib/airbrake/user_informer.rb:12:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/railties-4.2.4/lib/rails/engine.rb:518:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/railties-4.2.4/lib/rails/application.rb:165:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/urlmap.rb:66:in `block in call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/urlmap.rb:50:in `each'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/urlmap.rb:50:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/capybara-2.5.0/lib/capybara/server.rb:19:in `call'
          # /usr/local/rvm/gems/ruby-2.2.3@zeequote/gems/rack-1.6.4/lib/rack/handler/webrick.rb:88:in `service'

Finished in 3.45 seconds (files took 6.92 seconds to load)
2 examples, 1 failure

Failed examples:

rspec ./spec/features/contractors/signup_spec.rb:22 # Contractor Sign Up New contractor creates organization

Randomized with seed 43442

```

We'd love to continue using VCR to speed up our tests, but simply turning off this `around_http_request` wrapper we made for googlemaps API fixes our problem for now.

Here's also a related issue, but this is several years old, wasn't ever really solved, and the solution provided talks about nested cassettes, which we're not doing (to my knowledge).

https://github.com/vcr/vcr/issues/36

Thanks for your help!

**Comment by toobulkeh** [comment]
Also note that we do not have any Cassettes already generated. We had them at one point, but deleting them did not fix the issue. Running this doesn't generate any, so the first request never fires. If we're making calls within calls to the API that would lead me to think it's trying to use_casette in a nested manner.

**Comment by tombh** [comment]
I'm having the exact same issue. I haven't looked into it, but a quick workaround that works for me is to just monkey patch the offending line to only `dup` if non nil;

```ruby
module VCR
  def dup_context(context)
    {
      :turned_off => context[:turned_off],
      :ignore_cassettes => (context[:ignore_cassettes].dup if context[:ignore_cassettes]),
      :cassettes => context[:cassettes].dup
    }
  end
end
```

Simply paste that somewhere after VCR has been required. I've placed it before the configure block of my `spec_helper.rb` file.

**Comment by toobulkeh** [comment]
Thanks @tombh!

**Comment by mcfiredrill** [comment]
Related?
https://github.com/vcr/vcr/pull/516

**Comment by toobulkeh** [comment]
Looks like it. Thanks for the commit / merge! When does it go live?

Does this have to do with threadsafety overall? Sorry for my ignorance on threads.

**Comment by sbleon** [comment]
The fix in #516 fixes this issue, so I think this issue can be closed.

If anyone wants to use the fixed version before 3.0.1 is released, change your Gemfile to say:
`gem 'vcr', github: 'vcr/vcr'`

**Comment by krainboltgreene** [comment]
@sbleon I'll try to get a release out today, but until then please use a commit for your requirement!

**Comment by sbleon** [comment]
@krainboltgreene thanks for that suggestion. Looking forward to a new release!

**Comment by mcfiredrill** [comment]
Fixed in 3.0.1
